frontend/src/metabase/search/components/SearchResult.jsx

```diff
@@ -182,8 +188,10 @@ function DefaultResult({ result, options }) {
     <ResultLink to={result.getUrl()}>
       <Flex align="center">
         <ItemIcon item={result} />
-        <Title>{result.name}</Title>
-        {formatCollection(result.getCollection())}
+        <Box>
+          <Title>{result.name}</Title>
+          {formatCollection(result.getCollection())}
```

Comment:
This is on me, but looking at it with fresh eyes I guess `<CollectionName collection={result.getCollection()}>` would've been more idiomatic than the `formatCollection` function.

Reply:
Also ~on me, and probably for a separate PR, but I think relying on polymorphic behavior (e.g., `result.getUrl()` instead of `Urls.question(result.id)`) and an intelligent backend that returns `collection` and `context` information only when relevant so that we can do the conditional rendering is probably the direction to lean into so that we don't have all these sub-components to keep track of. `DashboardResult` could be replaced by `DefaultResult` in seconds (and already had the `<Box>` that this PR had to add), and I think converting the other ones would also be straightforward